### PR TITLE
Leaving the loading of extensions enabled might be a security risk

### PR DESCRIPTION
**Pull request by mgrojo**
Using `sqlite3_enable_load_extension` not only allows loading extensions
through the C-API but also through the SQL function `load_extension()`.
That might be a security risk if the user is unaware that executing an
SQL file can lead to native code execution and not only to database file
modification.

See issue #1551

### CODE REVIEW

File: src/sqlitedb.cpp [hunk 1]
```diff
@@ -1791,9 +1785,17 @@ bool DBBrowserDB::loadExtension(const QString& filePath)
...
 
+    // Enable extension loading
+    sqlite3_enable_load_extension(_db, 1);
+
     // Try to load extension
     char* error;
-    if(sqlite3_load_extension(_db, filePath.toUtf8(), nullptr, &error) == SQLITE_OK)
+    int result = sqlite3_load_extension(_db, filePath.toUtf8(), nullptr, &error);
+
+    // Disable extension loading (we don't want to left the possibility of calling load_extension() from SQL)
```
Comment (justinclift) [inline src/sqlitedb.cpp]: s/left/leave/ :smile:

Comment (mgrojo) [reply src/sqlitedb.cpp]: I wrote that? My eyes hurt :smile:

**Comment by MKleusberg** [comment]
I don't know about ```load_extension()``` but I remember that there are some users who don't use any of the GUI features we offer and only rely on the Execute SQL tab. No idea if they're loading extensions though but the preferable option for everybody might be a dialog which pops up and asks for an extra confirmation or something.

That said, I just tried to run this in the Execute SQL tab
```sql
load_extension('/path/to/extension.so');
```
and it returned a syntax error. In the SQL log and in the debugger I could see that SQLite splits this line into two statements: ```load_extension``` and ```('/path/to/extension.so');``` and the first one alone obviously is a syntax error. Does anybody else have the same problem?

**Comment by mgrojo** [comment]
The correct syntax would be:

`SELECT load_extension('/path/to/extension.so');`

Another option could be a preference setting for enabling the load of extensions in SQL. By default it should be disabled (following the design decision of SQLite itself).

**Comment by MKleusberg** [comment]
Haha, you're right! I forgot about the ```SELECT``` part :roll_eyes: And yeah, a config option would probably be the easiest solution. But one could still argue that you should get a warning when importing a malicious SQL file, even with the config option turned on. But before overthinking this, adding a config option might be a good first step :wink:

**Comment by mgrojo** [comment]
Ok, I'll add the setting to the branch.

**Comment by justinclift** [comment]
Excellent. :smile:

**Comment by mgrojo** [comment]
I just noticed that we haven't updated the translation files with the new strings present in this PR. Updating now might raise conflicts with the present or ongoing PR, so I don't know which is the best way to go.

**Comment by justinclift** [comment]
Update the strings in `master` and `v3.11.x` then let the translators know.

In theory, it shouldn't happen toooooo many times before release. :smile:

**Pull request merged**